src/service/kafkaRelayPlugin.ts

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: Apache-2.0
+import { Kafka, logLevel, type Producer } from 'kafkajs';
+import { additionalEnvironmentVariables, type Configuration } from '../config';
+import type { LoggerService } from '@tazama-lf/frms-coe-lib';
+import type { Apm } from '@tazama-lf/frms-coe-lib/lib/services/apm';
+import { validateProcessorConfig } from '@tazama-lf/frms-coe-lib/lib/config/processor.config';
+import * as fs from 'fs';
+import type { ITransportPlugin } from '@tazama-lf/frms-coe-lib/lib/interfaces/relay-service/ITransportPlugin';
+
+export default class KafkaRelayPlugin implements ITransportPlugin {
+  private readonly kafka: Kafka;
+  private producer?: Producer;
+  private loggerService?: LoggerService;
+  private apm?: Apm;
+  private readonly configuration: Configuration;
+  private readonly maxInFlight: number;
+
+  constructor() {
+    // Validate and load configuration
+    this.configuration = validateProcessorConfig(additionalEnvironmentVariables) as Configuration;
+
+    const isDev = !this.configuration.nodeEnv || this.configuration.nodeEnv === 'dev';
+
+    const ssl = isDev
```

Comment:
Give ssl a types for simpler expansion of connection options for developers
```suggestion
    const ssl: boolean | ConnectionOptions = isDev
```

ConnectionOptions from
`import type { ConnectionOptions } from 'tls';`